src/Core/Search/Filters.php

```diff
@@ -92,15 +92,15 @@ public function getOrderWay()
      */
     public function getOffset()
     {
-        return $this->get('offset');
+        return $this->getInt('offset') ?: null;
```

Comment:
Indeed it's not a BC break since it respects the `SearchCriteriaInterface` interface return type `int|null`